src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/ECDsaOpenSsl.cs

```diff
@@ -30,16 +31,9 @@ public ECDsaOpenSsl(SafeEvpPKeyHandle pkeyHandle)
                 throw new ArgumentException(SR.Cryptography_OpenInvalidHandle, nameof(pkeyHandle));
 
             ThrowIfNotSupported();
-            // If ecKey is valid it has already been up-ref'd, so we can just use this handle as-is.
-            SafeEcKeyHandle key = Interop.Crypto.EvpPkeyGetEcKey(pkeyHandle);
-            if (key.IsInvalid)
-            {
-                key.Dispose();
-                throw Interop.Crypto.CreateOpenSslCryptographicException();
-            }
 
-            _key = new ECOpenSsl(key);
-            KeySizeValue = _key.KeySize;
+            _key = new Lazy<SafeEvpPKeyHandle>(pkeyHandle.DuplicateHandle());
```

Comment:
The key should be checked for the correct algorithm/mode before being accepted.  (Which the current RSAOpenSsl does, but I think you deleted that code in this change)

Reply:
I've addressed this by checking key type